staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go

```diff
@@ -242,3 +248,54 @@ func generateKey(length int) (key []byte, err error) {
 
 	return key, nil
 }
+
+func validateEncryptedObject(o *kmstypes.EncryptedObject) error {
+	if o == nil {
+		return fmt.Errorf("encrypted object is nil")
+	}
+	if o.EncryptedDEK == nil {
+		return fmt.Errorf("encrypted dek is nil")
+	}
+	if o.EncryptedData == nil {
+		return fmt.Errorf("encrypted data is nil")
+	}
+	if err := validateKeyID(o.KeyID); err != nil {
+		return fmt.Errorf("failed to validate key id: %w", err)
+	}
+	if err := validateAnnotations(o.Annotations); err != nil {
+		return fmt.Errorf("failed to validate annotations: %w", err)
+	}
+	return nil
+}
+
+// validateAnnotations tests the following:
+//  1. The annotations key is what Kubernetes calls a "qualified name".
+//     This validation is the same as the label key name validation.
+//  2. The size of annotations keys + values is less than 32 kB.
+func validateAnnotations(annotations map[string][]byte) error {
```

Comment:
Can you enhance one of the other tests in this file that runs the complete transformer to have a case where the plugin sends an invalid annotation and confirm that that fails as expected?

Reply:
done! lmk what you think.